docs/source/examples/predict_bike_trips.ipynb

```diff
@@ -0,0 +1,414 @@
+    "    random_state=0,\n",
+    "    regression=True,\n",
+    ")\n",
+    "\n",
+    "X_train, X_holdout, y_train, y_holdout = splits"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Finding the Best Model\n",
+    "\n",
+    "Then, we run a search on the training set for the best machine learning model."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "automl = evalml.AutoMLSearch(\n",
+    "    problem_type='regression',\n",
+    "    objective='r2',\n",
+    "    random_state=0,\n",
+    ")\n",
+    "\n",
+    "automl.search(\n",
+    "    X=X_train,\n",
+    "    y=y_train,\n",
+    "    data_checks='disabled',\n",
+    "    show_iteration_plot=False,\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Once the search is complete, we can print out information about the best pipeline found, such as the parameters in each component. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "automl.best_pipeline.describe()\n",
+    "automl.best_pipeline.graph()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's score the model performance by evaluating predictions on the holdout set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "best_pipeline = automl.best_pipeline.fit(X_train, y_train)\n",
+    "\n",
+    "score = best_pipeline.score(\n",
+    "    X=X_holdout,\n",
+    "    y=y_holdout,\n",
+    "    objectives=['r2'],\n",
+    ")\n",
+    "\n",
+    "dict(score)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "From the pipeline, we can see which features are most important for predictions."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "feature_importance = best_pipeline.feature_importance\n",
+    "feature_importance = feature_importance.set_index('feature')['importance']\n",
+    "top_k = feature_importance.abs().sort_values().tail(20).index\n",
+    "feature_importance[top_k].plot.barh(figsize=(8, 8), fontsize=14, width=.7);"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Next Steps\n",
+    "\n",
+    "At this point, we have completed the machine learning application. We can revisit each step to explore and fine-tune with different parameters until the model is ready for deployment. For more information on how to work with the features produced by Featuretools, take a look at [the Featuretools documentation](https://docs.featuretools.com/). For more information on how to work with the models produced by EvalML, take a look at [the EvalML documentation](https://evalml.alteryx.com/)."
```

Comment:
I guess it's probably beyond the scope of this example, but at some point it might be good to demonstrate how we could put this into production. So, now that we have a model, how would we use it to make a prediction for the number of rides that will originate from a given station in the 13 hours window starting today?

Reply:
great idea! we should try to get this in in PR

Reply:
I may have not thought through the idea of making a prediction thoroughly enough before making that suggestion. Some of the data we have used to create features wouldn't be available if we are trying to predict the number of trips that will originate from a station starting now and looking 13 hours into the future.

For example, we wouldn't know the duration of the trips until or the destination station until after they happen, so we wouldn't want to use those columns for creating features. Similarly, we wouldn't know the gender of the riders, although I'm not sure that is being used 

We could know the start day/time and you might argue we could know the weather (or at least use a weather forecast).

So - not sure where that leaves us exactly, but we might need to restructure this a bit if we are going to include the prediction piece.

Reply:
Also - when we generate features for the future sample, we would want to also include the historical data that is available as well. Right now I think you are only generating features based on the data in the future sample dataset, but not using all the historical information that we have available - this has a pretty big impact on the values for several of the features that get created.

I think maybe the process would be to append the "future data" to the historical data, including only data for the columns that are known as the start of the prediction window, and then recompute the features on this complete set.

Reply:
@thehomebrewnerd good point, I think we can use the stop time as the index for generating features and concat the new entity set with the other one to combine historical and future data.

Reply:
For predictions, I can use a cutoff time relative to the latest information available in the dataset.

Reply:
@jeff-hernandez Still thinking through this a bit, but I don't think appending the future data is doing anything for us in the current notebook. In this case you are setting a cutoff value that is before the end time of all the future data, so none of the future data information is being used in making the features.  I think you would get the same predictions if you didn't append the future data, and just used the entire historical dataset.

Maybe we could structure the prediction problem like this: We know the last datetime in the historical data - say it is 2014-07-01 00:24:00 (stoptime). So, let's assme we want to predict the number of rides for each station for the 13 hours starting at 8am the next day. You could then use that same entitytset and to create a feature matrix setting the cutoff time to July 2 at 8am and that fm can be used for making the predictions.

Does that approach make sense or am I overlooking something (which is quite possible)?

Originally I was thinking we might want to use the future data to create some features that we would know before the trip is over - like week/is_weekend, as we could know those as we are picking the start of a window to use for prediction. We could also claim to know or estimate the temperature and events (from a weather forecast) The other information would not be known until the trip is over (duration, end station, etc). I think this would require us to use a secondary time index. This should allow us to create some features in the future data for the things we know, but not use the data that is unknown until the trip ends. This is more complicated to setup and explain - not sure if it's worth it or not for this example. Looking at the features we are generating, this might not add much value anyway, so maybe the first approach is fine for this example.

@kmax12 any thoughts you would like to add on this?

Reply:
@thehomebrewnerd good catch, the sample of the future data is currently too small to add new information. We can just move the cutoff time further out to 8am (2017-10-01 08:00:00). This will keep the step for combining the entity sets. I'm also fine with just removing the future data and using the existing entity set with a cutoff time relatitve to the latest historical data available. I think using a secondary time index might overcomplicate this example.